src/dataflow-types/src/client/controller/storage.rs

```diff
@@ -161,18 +168,35 @@ impl RustType<ProtoCollectionMetadata> for CollectionMetadata {
                 blob_uri: value.blob_uri,
                 consensus_uri: value.consensus_uri,
             },
-            timestamp_shard_id: value
-                .timestamp_shard_id
+            remap_shard: value
+                .remap_shard
                 .parse()
                 .map_err(TryFromProtoError::InvalidShardId)?,
-            persist_shard: value
-                .shard_id
+            data_shard: value
+                .data_shard
                 .parse()
                 .map_err(TryFromProtoError::InvalidShardId)?,
         })
     }
 }
 
+impl Codec for CollectionMetadata {
```

Comment:
We need to make a `impl<R: RustType> Codec for R` impl, but unfortunately coherence prevents us :(